docs/adr/0008_signing_verification.md

```diff
@@ -0,0 +1,623 @@
+```
+
+#### Via `--signer-spec`
+
+```yaml ./rsapss.yaml
+type: RSASSA-PKCS1-V1_5/v1alpha1
+```
+
+```shell
+ocm sign componentversion --signature mysig --signer-spec ./rsapss.yaml ghcr.io/open-component-model/ocm//ocm.software/ocm:0.17.0
+```
+
+### Verification
+
+#### Via `signing.configuration.ocm.software/v1alpha1` and `--verifier`
+
+```shell
+ocm verify componentversion --signature mysig --verifier rsapss ghcr.io/open-component-model/ocm//ocm.software/ocm:0.17.0
+```
+
+#### Via `--verifier-spec`
+
+```yaml ./rsapss.yaml
+type: RSASSA-PKCS1-V1_5/v1alpha1
+```
+
+```shell
+ocm verify componentversion --signature mysig --verifier-spec ./rsapss.yaml ghcr.io/open-component-model/ocm//ocm.software/ocm:0.17.0
+```
+
+### Credentials
+
+Generated Credential Consumer Identity for `GetSigningCredentialConsumerIdentity` or
+`GetVerifyingCredentialConsumerIdentity`:
+
+```yaml
+type: RSASSA-PKCS1-V1_5/v1alpha1
+name: "rsapss"
+```
+
+Returned Credentials from `credentials.config.ocm.software`:
+
+```yaml
+private_key_pem_file: "/path/to/myprivatekey.pem"
+public_key_pem_file: "/path/to/mypublickey.pem"
+```
+
+These credentials can be used for both signing and verifying, and can also be separately referenced when only signing or
+verifying.
+
+## Example Configuration (Sigstore)
+
+Sigstore requires a special external plugin that works as:
+
+- Signing Handler with `sign.sigstore.dev/v1alpha1`
+  - This handler can request credentials of type `OIDCIdentityToken/v1alpha1` to use for signing.
+- Verification Handler with `verify.sigstore.dev/v1alpha1`
+  - This handler can request credentials of type `PEM/v1alpha1` to use for offline verification.
+- Credential Graph Plugin
+  - This plugin can resolve credentials of type `SigstoreOIDC/v1` to use for signing.
+
+### Flow
+
+```mermaid
+sequenceDiagram
+    participant U as User
+    participant CLI as ocm CLI
+    participant Repo as OCM Repository
+    participant Plugin as Sigstore Plugin
+    participant Creds as Credential Graph
+    participant Browser as User Browser
+    participant OIDC as OIDC Provider
+    participant Fulcio as Fulcio CA
+    participant Rekor as Rekor Transparency Log
+
+    Note over U,CLI: Signing
+    U->>CLI: ocm sign componentversion --signer sigstore
+    CLI->>Repo: Download Component Descriptor
+    CLI->>CLI: Normalize + Compute Digest
+    CLI->>Plugin: GetSigningCredentialConsumerIdentity(config)
+    Plugin-->>CLI: Credential Identity (OIDCIdentityToken/v1alpha1)
+    CLI->>Creds: Resolve credentials for identity
+    Creds->>Plugin: Trigger SigstoreOIDC flow
+    Plugin->>Browser: Open auth URL (with PKCE)
+    U->>Browser: Authenticate + Consent
+    Browser->>Plugin: Redirect to localhost callback with auth code. Forward auth code to plugin callback server
+    Plugin->>OIDC: Exchange auth code for ID token
+    OIDC-->>Plugin: ID token
+    Plugin-->>Plugin: Store token for reuse (local cache)
```

Comment:
Would this be just for this flow and you'd have to reauthenticate on every consequite call? Or would we store it in under something like .config/ocm/cache with some key that lets us find it later on again?